projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -48,6 +48,9 @@ function jetpack_setup_blogging_prompt_response( $post_id ) {
 	if ( $prompt ) {
 		update_post_meta( $post_id, '_jetpack_blogging_prompt_key', $prompt_id );
 		wp_add_post_tags( $post_id, array( 'dailyprompt', "dailyprompt-$prompt_id" ) );
+		if ( ! empty( $prompt['bloganuary_id'] ) ) {
```

Comment:
This causes a fatal:

```
Fatal error: Uncaught Error: Cannot use object of type stdClass as array in wp-content/plugins/jetpack-dev/_inc/blogging-prompts.php:51
```

Reply:
hmmm for me it is saying that $prompt is an array, I logged `gettype($prompt)` and it returned `array`, and the check seemes to work, so I think there must be different cases perhaps? I tested uploading to my wpcom sandbox with `https://wordpress.com/post/bloggerforeru.wordpress.com?answer_prompt=1812` for a bloganuary prompt and `https://wordpress.com/post/bloggerforeru.wordpress.com?answer_prompt=1850` for a normal prompt. Do you remember how you were testing when you got the fatal? 🙏

I updated the code to use `array_key_exists` which will just return false if it is passed a stdClass. It worked (on my machine & test site) but I'd be interested if the code adds the tag for you or always returns false?

Thank you for testing this!

.

Reply:
> Do you remember how you were testing when you got the fatal?

I'm testing on a self-hosted site. In all my tests, I get an object back:

<img width="1483" alt="image" src="https://github.com/Automattic/jetpack/assets/426388/2b2745d9-0b83-400b-ab71-11063e6993ae">

<img width="1040" alt="image" src="https://github.com/Automattic/jetpack/assets/426388/121c510f-7d0e-49a9-a1a6-e36fef1f1040">

This should fix things:

```diff
diff --git a/projects/plugins/jetpack/_inc/blogging-prompts.php b/projects/plugins/jetpack/_inc/blogging-prompts.php
index f00541e6ed..88ec1604f0 100644
--- a/projects/plugins/jetpack/_inc/blogging-prompts.php
+++ b/projects/plugins/jetpack/_inc/blogging-prompts.php
@@ -48,8 +48,8 @@ function jetpack_setup_blogging_prompt_response( $post_id ) {
 	if ( $prompt ) {
 		update_post_meta( $post_id, '_jetpack_blogging_prompt_key', $prompt_id );
 		wp_add_post_tags( $post_id, array( 'dailyprompt', "dailyprompt-$prompt_id" ) );
-		if ( array_key_exists( 'bloganuary_id', $prompt ) ) {
-			wp_add_post_tags( $post_id, array( 'bloganuary', $prompt['bloganuary_id'] ) );
+		if ( property_exists( $prompt, 'bloganuary_id' ) ) {
+			wp_add_post_tags( $post_id, array( 'bloganuary', $prompt->bloganuary_id ) );
 		}
 	}
 }

```

Reply:
hmmm yes, good to test on both and bad to have different behavior on both! 😂 I'll look into whether I can standardize the `jetpack_get_blogging_prompt_by_id` function

Reply:
Ahh I think I have found it, in the implementation of proxy_request_to_wpcom, we parse the response with `json_decode`, which converts it to a stdClass object, I'll add a second parameter to turn it into an array and test that tomorrow, then we would be consistent. Thanks for helping me get set up with the jurassic.ninja site, it looks like as the code is now, I'm getting a fatal error from `property_exists`, but I'll update it tomorrow :). 

https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v3-endpoint-blogging-prompts.php#L536